json-cache/service-worker.js

```diff
@@ -0,0 +1,43 @@
+// [Working example](/serviceworker-cookbook/json-cache/).
+
+var CACHE_NAME = 'dependencies-cache';
+
+self.addEventListener('install', function(event) {
+  // Perform install step:  loading each item into cache
+  event.waitUntil(
+    caches.open(CACHE_NAME)
+      .then(function(cache) {
+        console.log('[install] Caches opened');
+
+        return fetch('files-to-cache.json').then(function(response) {
+          return response.json();
+        }).then(function(files) {
+          console.log('[install] Adding files from JSON file: ', files);
+          return cache.addAll(files);
+        });
+      })
+      .then(function() {
+        console.log('[install] All required resources have been cached, we\'re good!');
+      })
+  );
+});
+
+self.addEventListener('fetch', function(event) {
+  event.respondWith(
+    caches.match(event.request)
+      .then(function(response) {
+        // Cache hit - return response
+        if (response) {
+          console.log('[fetch] Returning from cache: ', event.request.url);
+          return response;
+        }
+
+        return fetch(event.request);
```

Comment:
I think it could be simpler to do
```js
return response || fetch(event.request);
```
And it's emphasyzes the fact that `fetch()` is like a fallback.